File: University/lib/Dto/Base/LookupDto.dart
import 'LookupHeaderDto.dart';

class LookupDto{
  String Entity = "";
  List<LookupHeaderDto>? Headers;
  List<dynamic>? Rows;
  int TotalPage = 0;
  int TotalRecord = 0;
  String WindowSize = "";

  LookupDto({
    this.Entity = "",
    this.Headers,
    this.Rows,
    this.TotalPage = 0,
    this.TotalRecord = 0,
    this.WindowSize = "",
  });

  factory LookupDto.fromJson(Map<String, dynamic> json) {
    return LookupDto(
      Entity: json["Entity"] ?? "",
      Headers: json["Headers"] != null ? (json["Headers"] as List).map((e) => LookupHeaderDto.fromJson(e)).toList() : null,
      Rows: json["Rows"] != null ? json["Rows"] as List : null,
      TotalPage: json["TotalPage"] ?? 0,
      TotalRecord: json["TotalRecord"] ?? 0,
      WindowSize: json["WindowSize"] ?? "",
    );
  }

  Map toMap() {
    var map = <String, dynamic>{};
    map["Entity"] = Entity;
    map["Headers"] = Headers != null ? Headers!.map((e) => e.toMap()).toList() : null;
    map["Rows"] = Rows;
    map["TotalPage"] = TotalPage;
    map["TotalRecord"] = TotalRecord;
    map["WindowSize"] = WindowSize;

    return map;
  }
}

File: University.Service/University.Dto/Base/LookupDto.cs
using System;
using System.Data;
using System.Collections.Generic;
using System.Runtime.Serialization;

namespace University.Dto.Base
{
    [DataContract]
    public class LookupDto
    {
        [DataMember]
        public string CODE { get; set; }
        [DataMember]
        public string DESC { get; set; }

        [DataMember]
        public string Entity { get; set; }
        [DataMember]
        public int TotalPage { get; set; }
        [DataMember]
        public int TotalRecord { get; set; }
        [DataMember]
        public List<LookupHeaderDto> Headers { get; set; }
        [DataMember]
        public DataTable Rows { get; set; }
        [DataMember]
        public string WindowSize { get; set; }
    }

    public class LookUpMappingDto : Mapper<LookupDto>
    {
        protected override LookupDto PopulateItem(IDataRecord dr)
        {
            LookupDto dto = new LookupDto();
            for (int i = 0; i < dr.FieldCount; i++)
            {
                MapProperty(dto, dr.GetName(i), dr[i]);
            }
            return dto;
        }
    }
}
